rotkehlchen/db/dbhandler.py

```diff
@@ -3511,3 +3521,61 @@ def get_associated_locations(self) -> Set[Location]:
         if cursor.fetchone()[0] >= 1:
             locations.add(Location.BALANCER)
         return locations
+
+    def add_history_events(self, history: List[HistoryBaseEntry]) -> None:
+        """Insert a list of history events in database. May raise:
+        - InputError if the events couldn't be stored in database
+        """
+        query_str = """INSERT INTO history_events(identifier, event_identifier, sequence_index,
+        timestamp,location, location_label, asset, amount, usd_value, notes,
+        type, subtype) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);"""
+        cursor = self.conn.cursor()
+        events = []
+        for event in history:
+            try:
+                events.append(event.serialize_for_db())
+            except DeserializationError as e:
+                self.msg_aggregator.add_error(
+                    f'Failed to process kraken event for database. {str(e)}',
+                )
+        try:
+            cursor.executemany(query_str, events)
+        except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
+            log.error(f'Failed to save history events: {history}')
+            raise InputError(f'Failed to save history events in database. {str(e)}') from e
+
+        self.update_last_write()
+
+    def delete_history_events(self, location: Location) -> None:
+        """
+        Deletes history entries following the criteria of the filter_query. May raise:
+        - DeserializationError if the location is not valid
+        """
+        # TODO: In the future this method should allow for more granularity in the delete query
+        query = 'DELETE FROM history_events WHERE location=?'
+        cursor = self.conn.cursor()
+        cursor.execute(query, (location.serialize_for_db(),))
+        cursor.execute(
+            f'DELETE FROM used_query_ranges WHERE name LIKE "{location}_history_events_%"',
+        )
+        self.update_last_write()
+
+    def get_history_events(self, filter_query: HistoryEventFilterQuery) -> List[HistoryBaseEntry]:
+        """
+        Get history events using the provided query filter. May raise:
+        - DeserializationError
+        """
+        query, bindings = filter_query.prepare(with_pagination=False)
```

Comment:
Why do you do with pagination = False here?

Reply:
This is used internally and not exposed to the API. Correct me if I should be using pagination internally too.